### ISSUE TITLE
Re-enabling a WP repo after previous deletion/recreation in forge leads to DB deadlock

### ISSUE THREAD

**Issue by pat-s** [woodpecker-ci/woodpecker#3020]
### Component

server

### Describe the bug

See https://codeberg.org/Codeberg-CI/feedback/issues/120

C/p issue description:

1. Create repo in Gitea
2. Enable it in WP
3. Delete repo in Gitea
4. Create repo with the same name in Gitea
5. Try to enable it again

-> error

When enabling a repo, there should be a check for an existing one in the DB and if one exists, it should be deleted. 

### System Info

```shell
`next-dfc2c265b1` (Dec 20th)
```

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't [already an issue](https://github.com/woodpecker-ci/woodpecker/issues) that reports the same bug to avoid creating a duplicate.
- [X] Checked that the bug isn't fixed in the `next` version already [https://woodpecker-ci.org/faq#which-version-of-woodpecker-should-i-use]
- [X] Check that this is a concrete bug. For Q&A join our [Discord Chat Server](https://discord.gg/fcMQqSMXJy) or the [Matrix room](https://matrix.to/#/#woodpecker:matrix.org).

**Comment by pat-s** [comment]
Yes, the same name - *before* deleting it in WP. And after the described state is present, it can't be deleted anymore because it can't be accessed as it's not enabled. Which means a user can't get out of this state alone until an admin deletes the WP repo.

As mentioned in related issues, WP doesn't know atm when the forge repo is deleted. So either

- the forge sends a webhook upon deletion, which also deletes the WP repo (new feature, probably complex)
- WP adds some guardrails upon activation, checking if the WP repo already exists and either accepts it (this would preserve the previous logs and state) or deletes it and starts fresh

**Comment by pat-s** [comment]
Ah yes, I haven't tried this (too simple 😅 ). Yet still, users also likely won't think of this and also don't know that they need to delete the WP repo to get out of this state. 

So something should be done on the WP side to prevent this state or give a meaningful error at least (atm only "an error occurred" is shown in the bottom right).

**Comment by 6543** [comment]
on enable we have to add an check that:
first check if we have in DB an repo with same owner&name ... if true -> try to sync via ForgeRemoteID if it has changed name ->  if yes update repo and move on ... if no update ID to the currentl one we try to activate and use the re-activate mode

**Comment by 6543** [comment]
<img width="528" height="318" alt="Image" src="https://github.com/user-attachments/assets/7d006096-9a3c-4e0c-966b-c0ea9c90b2d1" />

we now show a propper error message

**Comment by 6543** [comment]
I did add a pull that lets repo repair addopt such cases: #6370

so the user has the choise to either delete or repair the repo on woodpeckers side

**Comment by 6543** [comment]
I'll add one more UX dessicion by making the woodpecker repo accessible if you klick on the entry... witch is currently not the case

**Comment by 6543** [comment]
well I just will make a propper UX fix at https://github.com/woodpecker-ci/woodpecker/pull/6370

**Comment by 6543** [comment]
done: https://github.com/woodpecker-ci/woodpecker/pull/6370
